cloudinit/net/netplan.py

```diff
@@ -570,3 +571,32 @@ def available(target=None):
         if not subp.which(p, search=search, target=target):
             return False
     return True
+
+
+def wait_for_network() -> None:
+    """On networkd systems, manually wait for systemd-networkd-wait-online"""
+    # At the moment, this is only supported using the networkd renderer.
+    if network_manager.available():
+        LOG.debug("NetworkManager is enabled, skipping networkd wait")
+        return
+    wait_online_def: str = subp.subp(
+        ["systemctl", "cat", "systemd-networkd-wait-online.service"]
```

Comment:
1. Given that this function isn't really netplan-speciifc, it's more systemd-networkd specific, doesn't this helper function actually belong over in cloudinit/net/systemd.py? 
2.  If we happen to run in an environment which doesn't have the systemd-networkd-wait-online.service the systemctl cat will exit 1, we should figure out how we want to behave in that environment. Info or WARNING log and return without waiting?
3. If called on a system without netplan.io, do we want this function to log that condition and just return without waiting?